core/src/main/java/com/cprcrack/videowebview/VideoEnabledWebChromeClient.kt

```diff
@@ -0,0 +1,315 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.cprcrack.videowebview
+
+import android.media.MediaPlayer
+import android.view.SurfaceView
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebChromeClient
+import android.widget.FrameLayout
+import android.widget.VideoView
+import org.kiwix.kiwixmobile.core.main.KiwixWebView
+
+/**
+ * This class serves as a WebChromeClient to be set to a WebView, allowing it to play video.
+ * Video will play differently depending on target API level (in-line, fullscreen, or both).
+ *
+ * It has been tested with the following video classes:
+ * - android.widget.VideoView (typically API level <11)
+ * - android.webkit.HTML5VideoFullScreen$VideoSurfaceView/VideoTextureView (typically API level
+ * 11-18)
+ * - com.android.org.chromium.content.browser.ContentVideoView$VideoSurfaceView (typically API level
+ * 19+)
+ *
+ * Important notes:
+ * - For API level 11+, android:hardwareAccelerated="true" must be set in the application manifest.
+ * - The invoking activity must call VideoEnabledWebChromeClient's onBackPressed() inside of its own
+ * onBackPressed().
+ * - Tested in Android API levels 8-19. Only tested on http://m.youtube.com.
+ *
+ * @author Cristian Perez (http://cpr.name)
+ */
+
+open class VideoEnabledWebChromeClient constructor() : WebChromeClient(),
+    MediaPlayer.OnCompletionListener,
+    MediaPlayer.OnPreparedListener, MediaPlayer.OnErrorListener {
+  constructor(
+    nonVideoView: ViewGroup,
+    videoView: ViewGroup,
+    arg3: Any,
+    webView: KiwixWebView
+  ) : this()
+
+  interface ToggledFullScreenCallback {
+
+    fun toggledFullscreen(fullscreen: Boolean)
+  }
+
+  private var activityNonVideoView: View? = null
+  private var activityVideoView: ViewGroup? = null
+  private var loadingView: View? = null
+  private var webView: VideoEnabledWebView? = null
+
+  // Indicates if the video is being displayed using a custom view (typically full-screen)
+  private var isVideoFullscreen = false
+  private var videoViewContainer: FrameLayout? = null
+  private var videoViewCallback: CustomViewCallback? = null
+
+  private var toggledFullScreenCallback: ToggledFullScreenCallback? = null
+
+  // * Never use this constructor alone.
+  // * This constructor allows this class to be defined as an inline inner class in which the user can
+  // * override methods
+
+  // * Builds a video enabled WebChromeClient.
+  // *
+  // * @param activityNonVideoView A View in the activity's layout that contains every other view that
+  // * should be hidden when the video goes full-screen.
+  // * @param activityVideoView A ViewGroup in the activity's layout that will display the video.
+  // * Typically you would like this to fill the whole layout.
+
+  fun View?.videoEnabledWebChromeClient(activityVideoView: ViewGroup?) {
```

Comment:
Why did these constructors get turned into extension functions?

Did you convert this file using the automatic conversion `ctrl+alt+shift+k`?

Reply:
yeah i have done that i will change them manually and commit them